dateparser/search/__init__.py

```diff
@@ -1,63 +1,119 @@
-from dateparser.search.search import DateSearchWithDetection
+from dateparser.search.search import DateSearch
+from dateparser.conf import apply_settings
 
 
-_search_with_detection = DateSearchWithDetection()
+_search_dates = DateSearch()
```

Comment:
@gavishpoddar @noviluni hey folks, I'd like to clarify if we still want to try maintain backwards compatibility for things which are not documented but which are publicly exposed? If yes, then probably we would want to keep the same name of the class?

Reply:
Thanks for pointing this by this way things would be easier. So I am replacing `DateSearch` with `DateSearchWithDetection` so this PR will be backwards compatible. 😀